functions/src/submissions.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response } from 'express'
+import { collection, getDocs, query, where } from 'firebase/firestore'
+import { db } from './util'
+export async function getSubmissions(req: Request, res: Response) {
+  const uId = req.body.uid
+  const pIds: string[] = req.body.problemIds
+  const isBrief = req.body.isBrief
+  try {
+    const allSubmissions: object[] = []
+
+    const add = (submission: object) => {
+      return new Promise((resolve, reject) => {
+        if (submission !== null) {
+          allSubmissions.push(submission)
+          resolve(allSubmissions)
+        } else {
+          reject(new Error('invalid'))
+        }
+      })
+    }
+
+    await Promise.all(
+      pIds.map(async (pId) => {
+        const submissionList: object[] = []
+        const submDb = collection(db, 'Submissions')
+        const queries = query(
+          submDb,
+          where('problem_id', '==', pId),
+          where('uid', '==', uId),
+        )
+
+        const docs = await getDocs(queries)
+        docs.forEach((doc) => submissionList.push(doc.data()))
+
+        let problem = {}
+        if (isBrief) {
+          const isSubmitted = submissionList.length > 0
+          let isACed = false
+          submissionList.forEach((submission) => {
+            const submissionJson = JSON.parse(JSON.stringify(submission))
+            if (submissionJson.verdict == 3) isACed = true
```

Comment:
Are the verdict IDs listed somewhere with their meanings?

Reply:
they're in judge0, can get them from an api call, not sure how we want to communicate them between since they're stored in a map

Reply:
https://ce.judge0.com/statuses